pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,34 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MultiClusterServiceSpec `json:"spec"`
+}
+
+type MultiClusterServiceSpec struct {
+	ServiceName string                    `json:"service_name"`
+	Items       []MultiClusterServiceInfo `json:"items"`
+}
+
+type MultiClusterServiceInfo struct {
+	ClusterID     string `json:"cluster_id"`
+	ClusterDomain string `json:"cluster_domain"`
+	ServiceIP     string `json:"service_ip"`
```

Comment:
Are those the CRDs created by lighthouse and distributed to the clusters?

It would be great to add documentation about how will they work and why are those fields selected, bear in mind that we will sooner or later need to bring this discussion up to the SIGs and agree on how this works.

Reply:
Yes those will be created and distributed by lighthouse , have added more details to the design document in the preferred solution section.

[1]https://docs.google.com/document/d/1TDTzJxAX7_aBzrd6I1JaK5TzVhGnSPPTk2HFQeqHst4/edit#heading=h.t4qtmr0x20x

Reply:
But I mean, documenting it here, in the codebase :)

Reply:
I mean, adding comments, like this example: 
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/networking/types.go#L38

Explaining what are the objects used for, exactly...
and what is each field for. In godoc format, so we would have automatic documentation for those CRDs.

Reply:
> Explaining what are the objects used for, exactly...
> and what is each field for. In godoc format, so we would have automatic documentation for those CRDs.

+1